### PR TITLE
feat(core): use stricter types

### PR DESCRIPTION
**Pull request by gershon**
## Description

Avoid BigNumberish and use stricter types.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature (`feat:`)
- [ ] 🐛 Bug Fix (`fix:`)
- [ ] 📝 Documentation Update (`docs:`)
- [ ] 🎨 Style (`style:`)
- [ ] 🧑‍💻 Code Refactor (`refactor:`)
- [ ] 🔥 Performance Improvements (`perf:`)
- [ ] ✅ Test (`test:`)
- [ ] 🤖 Build (`build:`)
- [ ] 🔁 CI (`ci:`)
- [ ] 📦 Chore (`chore:`)
- [ ] ⏩ Revert (`revert:`)
- [ ] 🚀 Breaking Changes (`BREAKING CHANGE:`)

### CODE REVIEW

**Review by kwiss — COMMENTED** [review]
I think it miss the changeset

File: apps/example/package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "starknet": "6.6.6",
     "tailwind-merge": "^2.1.0",
     "tailwindcss-animate": "^1.0.7",
+    "viem": "^2.12.1",
     "web3": "^4.4.0",
```
Comment (kwiss) [inline apps/example/package.json:52]: web3 should be remove now

**Review by kwiss — CHANGES_REQUESTED** [review]

File: apps/example/package.json (hunk 1 above)
Comment (gershon) [reply apps/example/package.json:52]: Web3 is still used in apps/example, we might fully remove it via another PRs.

File: packages/core/src/actions/order/fulfillAuction.ts [hunk 2]
```diff
@@ -32,19 +32,19 @@ const fulfillAuction = async (
   const chainId = await config.starknetProvider.getChainId();
 
   const fulfillInfo: FulfillInfo = {
-    order_hash: fulfillAuctionInfo.orderHash,
-    related_order_hash: new CairoOption<BigNumberish>(
+    orderHash: fulfillAuctionInfo.orderHash,
+    relatedOrderHash: new CairoOption<BigNumberish>(
```
Comment (kwiss) [inline packages/core/src/actions/order/fulfillAuction.ts:36]: BigNumberish

File: packages/core/src/actions/order/fulfillListing.ts [hunk 3]
```diff
@@ -36,16 +36,16 @@ const fulfillListing = async (
   const { starknetAccount, fulfillListingInfo, approveInfo } = parameters;
   const chainId = await config.starknetProvider.getChainId();
   const fulfillInfo: FulfillInfo = {
-    order_hash: fulfillListingInfo.orderHash,
-    related_order_hash: new CairoOption<BigNumberish>(CairoOptionVariant.None),
+    orderHash: fulfillListingInfo.orderHash,
+    relatedOrderHash: new CairoOption<BigNumberish>(CairoOptionVariant.None),
```
Comment (kwiss) [inline packages/core/src/actions/order/fulfillListing.ts:40]: BigNumberish

File: packages/core/src/actions/order/fulfillOffer.ts [hunk 4]
```diff
@@ -36,16 +36,16 @@ const fulfillOffer = async (
   const { starknetAccount, fulfillOfferInfo, approveInfo } = parameters;
   const chainId = await config.starknetProvider.getChainId();
   const fulfillInfo: FulfillInfo = {
-    order_hash: fulfillOfferInfo.orderHash,
-    related_order_hash: new CairoOption<BigNumberish>(CairoOptionVariant.None),
+    orderHash: fulfillOfferInfo.orderHash,
+    relatedOrderHash: new CairoOption<BigNumberish>(CairoOptionVariant.None),
```
Comment (kwiss) [inline packages/core/src/actions/order/fulfillOffer.ts:40]: BigNumberish

File: packages/core/src/types/index.ts [hunk 5]
```diff
@@ -1,108 +1,107 @@
-import type { BigNumberish, CairoOption, Uint256 } from "starknet";
+import type { BigNumberish, CairoOption, constants, Uint256 } from "starknet";
```
Comment (kwiss) [inline packages/core/src/types/index.ts:1]: BigNumberish

File: packages/core/src/types/index.ts [hunk 6]
```diff
@@ -1,108 +1,107 @@
...
+  tokenAddress: string;
   tokenId: CairoOption<Uint256>;
   quantity: Uint256;
   startAmount: Uint256;
   endAmount: Uint256;
-  salt: BigNumberish;
+  salt: number;
   startDate: number;
   endDate: number;
   additionalData: BigNumberish[];
```
Comment (kwiss) [inline packages/core/src/types/index.ts:23]: BigNumberish

File: packages/core/src/types/index.ts [hunk 7]
```diff
@@ -1,108 +1,107 @@
...
 export type FulfillInfo = {
-  order_hash: BigNumberish;
-  related_order_hash: CairoOption<BigNumberish>;
-  fulfiller: BigNumberish;
-  token_chain_id: BigNumberish;
-  token_address: BigNumberish;
-  token_id: CairoOption<Uint256>;
-  fulfill_broker_address: BigNumberish;
+  orderHash: bigint;
+  relatedOrderHash: CairoOption<BigNumberish>;
```
Comment (kwiss) [inline packages/core/src/types/index.ts:76]: BigNumberish

**Review by kwiss — APPROVED** [review]

**Pull request merged**